core/src/main/java/org/testcontainers/utility/DockerImageName.java

```diff
@@ -165,6 +165,9 @@ public void assertValid() {
         if (!versioning.isValid()) {
             throw new IllegalArgumentException(versioning + " is not a valid image versioning identifier (in " + rawName + ")");
         }
+        if (!rawName.contains("/") && rawName.split(":").length > 2) {
```

Comment:
Since this PR is specifically looking for an unallowed `:` in the repo name, let's increase the semantics of the validation and make use of the already existing `REPO_NAME.matcher` here, instead of this borad check on the raw name.

Reply:
The problem is, that at this stage the last part of the name is already lost.
`repo:repo:version` becomes:
```
repository = "repo"
versioning = TagVersioning("repo")
```
the only information where the error can be detected is the `rawName`.
I did not want to introduce the check in the constructor because I think that this method should have the responsibility for the assertion of the right name.